clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -556,46 +562,120 @@ void SemaHLSL::handleShaderAttr(Decl *D, const ParsedAttr &AL) {
     D->addAttr(NewAttr);
 }
 
-void SemaHLSL::handleResourceClassAttr(Decl *D, const ParsedAttr &AL) {
-  if (!AL.isArgIdent(0)) {
-    Diag(AL.getLoc(), diag::err_attribute_argument_type)
-        << AL << AANT_ArgumentIdentifier;
-    return;
-  }
+bool clang::CreateHLSLAttributedResourceType(
+    Sema &S, QualType Wrapped, llvm::SmallVector<const Attr *> &AttrList,
```

Comment:
We should be able to use an `ArrayRef<const Attr *> AttrList` instead of the SmallVector reference here since we just iterate over it. Note that if we did need the SmallVector API it would be better to use `SmallVectorImpl` instead since it doesn't encode a specific size.

Reply:
Cool! I did not know SmallVector implicitly converts to ArrayRef when the argument requires it!